### PR TITLE
Fix compatibility with puppet-network

### PR DESCRIPTION
**Pull request by pedmon**
See the same issue in the Infiniband modue: https://github.com/treydock/puppet-infiniband/issues/35  Below is not a fix but rather a start to one. Unfortunately the current puppet-network module (https://forge.puppet.com/modules/puppet/network) does not have the fields necessary to make this work yet. My question is how do you want to proceed?

### CODE REVIEW

**Comment by pedmon** [comment]
I found a way in puppet-network to pass arbitrary variables. I think this should cover most of it.

File: manifests/interface.pp [hunk 1]
```diff
@@ -79,45 +79,52 @@
...
+      network_config { $ifname:
         ensure               => $ensure,
-        enable               => $enable,
         onboot               => $onboot,
-        type                 => 'InfiniBand',
         master               => $name,
         slave                => 'yes',
-        nm_controlled        => $_nm_controlled,
         mtu                  => $mtu,
-        options_extra_redhat => $options_extra_redhat,
```
Comment (treydock) [inline manifests/interface.pp:91]: `$options_extra_redhat` needs to be kept but could likely just be moved directly into the `options` hash by setting `CONNECTED_MODE => $connected_mode`.

**Comment by treydock** [comment]
These places need updates:

https://github.com/treydock/puppet-module-mofed/blob/772c1ec3ce02f332d75fb33d048534a53093dc5a/.fixtures.yml#L15-L17

https://github.com/treydock/puppet-module-mofed/blob/772c1ec3ce02f332d75fb33d048534a53093dc5a/metadata.json#L28-L29

File: manifests/interface.pp (hunk 1 above)
Comment (pedmon) [reply manifests/interface.pp:91]: Okay, I've added those to the options hash.

**Comment by pedmon** [comment]
I've made the requested changes.  I saw that you just committed a host of other changes but it looks like these are mutually exclusive. Still let me know if you need me to tweak anything else.

**Pull request closed without merging**